docker/Dockerfile.thumbv6m-none-eabi

```diff
@@ -10,14 +10,22 @@ RUN /cmake.sh
 COPY xargo.sh /
 RUN /xargo.sh
 
-COPY qemu.sh /
 RUN apt-get update && apt-get install --assume-yes --no-install-recommends \
     gcc-arm-none-eabi \
-    libnewlib-arm-none-eabi && \
-    /qemu.sh arm
+    libnewlib-arm-none-eabi \
+    libstdc++-arm-none-eabi-newlib
+
+COPY qemu.sh /
+RUN /qemu.sh arm
+
+COPY newlib.cmake /opt/newlib.cmake
+ENV CMAKE_TOOLCHAIN_FILE=/opt/newlib.cmake
+ENV CROSS_NEWLIB_OBJECT_FLAGS="-ffunction-sections -fdata-sections -mthumb -march=armv6s-m"
 
-ENV QEMU_CPU=cortex-m3 \
+ENV QEMU_CPU=cortex-m1 \
```

Comment:
`cortex-m3` is an ARMv7-M target, `cortex-m1` is the proper CPU here for ARMv6-M (see [ARM Cortex-M](https://en.wikipedia.org/wiki/ARM_Cortex-M#Cortex-M1)).